libs/labelbox/src/labelbox/schema/user_group.py

```diff
@@ -0,0 +1,523 @@
+from labelbox.pydantic_compat import BaseModel
+from labelbox.schema.user import User
+from labelbox.schema.project import Project
+from labelbox.exceptions import UnprocessableEntityError, InvalidQueryError
+
+
+class UserGroupColor(Enum):
+    """
+    Enum representing the colors available for a group.
+
+    Attributes:
+        BLUE (str): Hex color code for blue (#9EC5FF).
+        PURPLE (str): Hex color code for purple (#CEB8FF).
+        ORANGE (str): Hex color code for orange (#FFB35F).
+        CYAN (str): Hex color code for cyan (#4ED2F9).
+        PINK (str): Hex color code for pink (#FFAEA9).
+        LIGHT_PINK (str): Hex color code for light pink (#FFA9D5).
+        GREEN (str): Hex color code for green (#3FDC9A).
+        YELLOW (str): Hex color code for yellow (#E7BF00).
+        GRAY (str): Hex color code for gray (#B8C4D3).
+    """
+    BLUE = "9EC5FF"
+    PURPLE = "CEB8FF"
+    ORANGE = "FFB35F"
+    CYAN = "4ED2F9"
+    PINK = "FFAEA9"
+    LIGHT_PINK = "FFA9D5"
+    GREEN = "3FDC9A"
+    YELLOW = "E7BF00"
+    GRAY = "B8C4D3"
+
+
+class UserGroupUser(BaseModel):
+    """
+    Represents a user in a group.
+
+    Attributes:
+        id (str): The ID of the user.
+        email (str): The email of the user.
+    """
+    id: str
+    email: str
+
+    def __hash__(self):
+        return hash((self.id))
+
+    def __eq__(self, other):
+        if not isinstance(other, UserGroupUser):
+            return False
+        return self.id == other.id
+
+
+class UserGroupProject(BaseModel):
+    """
+    Represents a project in a group.
+
+    Attributes:
+        id (str): The ID of the project.
+        name (str): The name of the project.
+    """
+    id: str
+    name: str
+
+    def __hash__(self):
+        return hash((self.id))
+
+    def __eq__(self, other):
+        """
+        Check if this GroupProject object is equal to another GroupProject object.
+
+        Args:
+            other (GroupProject): The other GroupProject object to compare with.
+
+        Returns:
+            bool: True if the two GroupProject objects are equal, False otherwise.
+        """
+        if not isinstance(other, UserGroupProject):
+            return False
+        return self.id == other.id
+
+
+class UserGroup:
+    """
+    Represents a user group in Labelbox.
+
+    Args:
+        client (Client): The Labelbox client.
+        **kwargs: Additional keyword arguments for initializing the UserGroup object.
+
+    Attributes:
+        _id (str): The ID of the user group.
+        _name (str): The name of the user group.
+        _color (UserGroupColor): The color of the user group.
+        _users (Set[Union[UserGroupUser, User]]): The set of user IDs in the user group.
+        _projects (Set[Union[UserGroupProject, Project]]): The set of project IDs in the user group.
+        _client (Client): The Labelbox client.
+    """
+    _id: str
+    _name: str
+    _color: UserGroupColor
+    _users: Set[Union[UserGroupUser, User]]
```

Comment:
do not see why we include actual `User` and `Project` here and below, since when we get the data from the server we always make them `UserGroupUser`, etc.

Reply:
No reason, just for convenience if people are iteracting with full user / project objects, I'll remove